algorithms/vuzzer/vuzzer_mutator.cpp

```diff
@@ -359,13 +366,39 @@ void VUzzerMutator::IntSlide() {
         outbuf = new_buf;
     }
 }
+    
+void VUzzerMutator::OverwriteDictWord() {
+    const auto dict_end = std::partition_point(
+      state.extras.begin(),
+      state.extras.end(),
+      [len=len]( const auto &v ) {
+        return v.data.size() <= len;
+      }
+    );
+    const auto dict_size = std::distance( state.extras.begin(), dict_end );
+    const unsigned int dict_index = rand() % dict_size;
+    const auto &word = state.extras[ dict_index ].data;
+    const auto word_len = word.size();
+    const int start_pos = ( len == word_len ) ? 0 : ( rand() % ( len - word_len ) );
```

Comment:
```suggestion
    const int start_pos = rand() % ( len - word_len + 1);
```

Consider the case where overwrite a part of `"AA"` with `"B"`: there are two cases `"AB"` and `"BA"` .

Reply:
fixed

Reply:
fixed